apache/maven
# [Experiment] Maven w/ transport-http

This PR creates maven distro (kept up to date with master) that contains latest (locally built) maven-resolver along w/ transport-http and transport-wagon. Also, the build banishes wagon from Maven, but it is still present in lib of distro as maven-compat still uses it.

Try this build locally, to build it:
* build/install locally maven-resolver master
* build locally distro using this PR
* deploy locally the built maven

To test it:
* try to locally build master (due resolver SNAPSHOT dependency) of maven using vanilla distro and this distro (with or without local MRM). To notice the speed difference between two transports, always use empty local repo (to make Maven pull all).
* try out the "provided" checksum features (provided checksums are in this PR, but to build this PR you must set maven dependency to latest released resolver). Using `.mvn/checksums-bad` will fail the build, while `.mvn/checksums-good` will make build pass when plexus-utils:3.3.0 is being fetched.

(Don't forget to disable RAT if you use in-dir local repo)

Example of "provided" use:
```
~/bin/maven/apache-maven-4.0.0-alpha-1-SNAPSHOT-transport-http/bin/mvn clean install  -Daether.artifactResolver.providedChecksumsSource.file.baseDir=/PATH/TO/maven/.mvn/checksums-bad -DlocalRepository=localRepository -Drat.skip=true
```

To switch transport to wagon (default is transport-http) add this parameter:
```
-Daether.priority.WagonTransporterFactory=100
```
